x/qgb/types/validator.go

```diff
@@ -0,0 +1,180 @@
+package types
+
+import (
+	"bytes"
+	math "math"
+	"sort"
+
+	"github.com/ethereum/go-ethereum/common"
+
+	stakingtypes "github.com/cosmos/cosmos-sdk/x/staking/types"
+
+	sdkerrors "github.com/cosmos/cosmos-sdk/types/errors"
+)
+
+// ToInternal transforms a BridgeValidator into its fully validated internal type.
+func (b BridgeValidator) ToInternal() (*InternalBridgeValidator, error) {
+	return NewInternalBridgeValidator(b)
+}
+
+// BridgeValidators is the sorted set of validator data for Ethereum bridge MultiSig set.
+type BridgeValidators []BridgeValidator
+
+func (b BridgeValidators) ToInternal() (*InternalBridgeValidators, error) {
+	ret := make(InternalBridgeValidators, len(b))
+	for i := range b {
+		ibv, err := NewInternalBridgeValidator(b[i])
+		if err != nil {
+			return nil, sdkerrors.Wrapf(err, "member %d", i)
+		}
+		ret[i] = ibv
+	}
+	return &ret, nil
+}
+
+// Bridge Validator but with validated EthereumAddress.
+type InternalBridgeValidator struct {
```

Comment:
[comment]
we might want to change the name of these to avoid confusion, but its a very low priority since they're strictly internal and we inherited them from the gravity module

Reply:
Do you have any other name in mind?  I guess there are lot of structs that  can be renamed to self  reflect what  they do. Do we  need an issue?

Reply:
no, I definitely don't think we have to change this in this PR or even create an issue. It was just a comment to see if something else pops up.